person.rb

```diff
@@ -1,9 +1,11 @@
-class Person
-  attr_accessor :name, :age
-  attr_reader :id
+require_relative '../decorators/nameable'
 
-  def initialize(age, name: 'unknown', parent_permission: true)
-    @id = 0
+class Person < Nameable
+  attr_accessor :name, :age, id
+
+  def initialize(name: 'Unknown', age: nil, parent_permission: true)
```

Comment:
- [ ] Great job so far making some of the changes, ie Person class inherits from Nameable class. Please, like the previous reviewer also clearly noted, the person class should have a method `correct_name`. Note that this method is needed by your decorators to work well. eg
```ruby
def correct_name
       name
    end
```